model/src/main/java/org/projectnessie/api/TreeApi.java

```diff
@@ -129,38 +144,47 @@ LogResponse getCommitLog(
       @Valid @NotNull CommitLogParams params)
       throws NessieNotFoundException;
 
-  /** Update a reference's HEAD to point to a different commit. */
+  /**
+   * Returns a set of content differences between two given references.
+   *
+   * @param params The {@link DiffParams} that includes the parameters for this API call.
+   * @return A set of diff values that show the difference between two given references.
+   */
+  DiffResponse getDiff(@Valid @NotNull DiffParams params) throws NessieNotFoundException;
```

Comment:
Let's make `DiffResponse` pageable in v2 (just adding the request/response parameters now)

Reply:
#5423 - I'd like to keep this PR limited to v2 API that cover the same functionality that also exists in v1.